src/button/Button.tsx

```diff
@@ -0,0 +1,121 @@
+import "./_button.scss";
+
+import React, {useEffect, useRef} from "react";
+import classNames from "classnames";
+
+import Spinner from "../spinner/Spinner";
+
+export interface ButtonProps {
+  testid: string;
+  children: React.ReactNode;
+  onClick?: React.ReactEventHandler<HTMLButtonElement>;
+  onMouseOver?: React.ReactEventHandler<HTMLButtonElement>;
+  onMouseDown?: React.ReactEventHandler<HTMLButtonElement>;
+  onMouseUp?: React.ReactEventHandler<HTMLButtonElement>;
+  onFocus?: React.ReactEventHandler<HTMLButtonElement>;
+  onBlur?: React.ReactEventHandler<HTMLButtonElement>;
+  id?: string;
+  lang?: string;
+  type?: "button" | "submit" | "reset";
+  shouldDisplaySpinner?: boolean;
+  isDisabled?: boolean;
+  shouldPreventDefault?: boolean;
+  shouldStopPropagation?: boolean;
+  shouldFocus?: boolean;
+  ariaLabel?: string;
+  customClassName?: string;
+  tabIndex?: number;
+}
+
+function Button({
+  testid,
+  type = "button",
+  id,
+  lang,
+  onClick,
+  children,
+  customClassName,
+  shouldPreventDefault = true,
+  shouldStopPropagation = true,
+  shouldFocus,
+  shouldDisplaySpinner,
+  ariaLabel,
+  isDisabled,
+  onMouseOver,
+  onMouseDown,
+  onMouseUp,
+  onFocus,
+  onBlur,
+  tabIndex
+}: ButtonProps) {
+  const isButtonDisabled = Boolean(isDisabled || shouldDisplaySpinner);
+  const containerClassName = classNames("button", customClassName, {
+    inactive: isButtonDisabled,
+    pending: shouldDisplaySpinner
+  });
+  const buttonRef = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    if (shouldFocus && buttonRef.current) {
+      buttonRef.current.focus();
+    }
+  }, [shouldFocus]);
+
+  return (
+    <button
+      ref={buttonRef}
+      id={id}
+      data-testid={testid}
+      tabIndex={tabIndex}
+      className={containerClassName}
+      type={type}
+      lang={lang}
+      onClick={handleClick}
+      onMouseOver={handleMouseOver}
+      onMouseDown={onMouseDown}
+      onMouseUp={onMouseUp}
+      onFocus={handleFocus}
+      onBlur={onBlur}
+      disabled={isButtonDisabled}
+      aria-label={ariaLabel}>
+      {shouldDisplaySpinner ? (
+        <Spinner
+          customClassName={"button-spinner"}
+          aria-label={"Button spinner visible. Button inactivated."}
+        />
+      ) : (
+        children
+      )}
```

Comment:
Should we hide the button content while displaying spinner? I think we can show spinner next to content (_as in the screenshot_), or make this behavior optional with a prop.
![Screen Shot 2020-04-28 at 11 12 36](https://user-images.githubusercontent.com/22727931/80463900-a0179100-8941-11ea-9c2a-c03b41a28595.png)

Reply:
Thanks! I think making it optional would be good